### DISCUSSION TITLE
Better notifications when a merge commit is added to a PR

### DISCUSSION THREAD

**Discussion by pxeger** [community/community#9862] [Pull Requests]
When you're subscribed to a PR and get notifications about commits, if someone adds a merge commit to the PR's branch, then you get a notification about it. That's fine. But when you click the notification, it takes you to the Changes page with the "changes since you last viewed", where there are none; only a message about a merge commit:

![2022-01-08_16-24](https://user-images.githubusercontent.com/67427173/148651951-6379738a-359f-4a96-be87-83f782c43d51.png)

Without context, this is a bit confusing. I've only just clicked on the notification, so I don't know what the merge commit is. Only when going back to the Conversation tab of the PR does it become clear what happened:

I think the best solution would be to reword this message:

> # The merge commit was added into this branch cleanly.
> There are no new changes to show, but you can still [view the diff](https://example.com).

into something like:

> # A merge commit was added into this branch
> It was applied cleanly, so there are no new changes to show. You can still [view the diff](https://example.com).

This way it covers more circumstances more clearly, while keeping the same meaning.

**Comment by uriklar** [comment]
I'd even prefer (possibly as a setting?) not to be notified about these cases at all. Or, maybe they could be marked somehow in the notifications list so I can handle them from there more easily. I mean, 100% of the time this type of notification doesn't require anything from me. So if it's just FYI maybe there's a way to know that without opening the notification.
